event-bus/api/src/main/java/org/apache/james/events/EventDeadLetters.java

```diff
@@ -93,6 +94,10 @@ public String asString() {
 
     Mono<Event> failedEvent(Group registeredGroup, InsertionId failDeliveredInsertionId);
 
+    default Mono<Event> failedEvent(Group registeredGroup, InsertionId failDeliveredInsertionId, Map<Group, EventSerializer> mapGroupToEventSerializer) {
```

Comment:
No we shall inject the event busses into event dead letter IMP we do not need this extra interface method.

Reply:
Circular dependency injection error occurred when I tried to inject event busses into event dead letter

Reply:
Bad answer.

Then we 'just need one more level of indirection...

How about a `EventBusses` interface that would abstract this away tfor us?

Reply:
Could you give me some more detail? I cannot figure out the way to fix the circular dependency.